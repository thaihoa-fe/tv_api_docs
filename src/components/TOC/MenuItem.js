import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { Link as GatsbyLink } from 'gatsby';
import { MdChevronRight } from 'react-icons/md';

const topLinkStyles = css`
  &.active {
    background: #fff;
    border-color: #e6ecf1;
  }
`;
const childLinkStyles = css`
  border-left: 1px solid #e6ecf1;
`;

const topTextStyles = css`
  color: #5c6975;

  .active > & {
    color: #3884ff;
  }
`;

const ExpandIcon = styled(MdChevronRight)`
  color: #9daab6;
  font-size: 18px;
  transition: all 0.2s linear;
  .active & {
    transform: rotate(90deg);
  }
  :hover & {
    color: #5c6975;
  }
`;

const Link = styled(GatsbyLink)`
  padding: 7px 24px 7px 16px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-decoration: none;
  border: 1px solid transparent;
  border-width: 1px 0px 1px 1px;
  border-style: solid solid solid;
  border-color: transparent transparent transparent;
  margin-left: ${props => (props.level - 1) * 16}px;
  ${props => (props.level === 1 ? topLinkStyles : null)}
  ${props => (props.level !== 1 ? childLinkStyles : null)}
  &:hover {
    background-color: rgb(230, 236, 241);
    border-color: rgb(230, 236, 241) rgb(230, 236, 241) rgb(230, 236, 241);
  }
`;

const Text = styled.span`
  font-size: 14px;
  line-height: 1.5;
  font-weight: 500;
  color: #9daab6;

  ${props => (props.level === 1 ? topTextStyles : null)}
`;

function MenuItem({ level, text, path, onClick }) {
  return (
    <Link onClick={onClick} level={level} to={path} title={text} activeClassName="active">
      <Text level={level}>{text}</Text>
      {level === 1 && <ExpandIcon />}
    </Link>
  );
}

MenuItem.propTypes = {
  text: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  level: PropTypes.number.isRequired,
  onClick: PropTypes.func,
};

MenuItem.defaultProps = {
  onClick: () => {},
};

export default MenuItem;
