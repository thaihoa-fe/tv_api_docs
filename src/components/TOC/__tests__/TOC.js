import React from 'react';
import renderer from 'react-test-renderer';
import MenuItem from '../MenuItem';
import { PureTOC as TOC } from '..';
import categories from './__fixtures__/categories.json';

describe('TOC', () => {
  const history = {
    location: {
      pathname: '/about_trusting_social/',
    },
  };

  it('renders correctly with empty categories', () => {
    const tree = renderer.create(<TOC history={history} />).toJSON();
    expect(tree).toBe(null);
  });

  it('renders correctly with categories', () => {
    const testRenderer = renderer.create(<TOC history={history} categories={categories} />);
    const tree = testRenderer.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with correct quantity of MenuItem with categories', () => {
    const matchItems = categories.filter(
      item => item.path === history.location.pathname || item.depth === 1
    );
    const testInstance = renderer.create(<TOC history={history} categories={categories} />).root;
    expect(testInstance.findAllByType(MenuItem)).toHaveLength(matchItems.length);
  });
});
