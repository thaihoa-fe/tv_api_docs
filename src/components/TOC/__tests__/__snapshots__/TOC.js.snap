// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOC renders correctly with categories 1`] = `
<div
  className="sc-bxivhb kiRNxH"
>
  <a
    className="sc-bwzfXH eevfLr"
    href="/about_trusting_social/"
    level={1}
    onClick={[Function]}
    title="Introduction"
  >
    <span
      className="sc-htpNat eqFeY"
    >
      Introduction
    </span>
    <svg
      className="sc-bdVaJa jjLnhl"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </a>
  <a
    className="sc-bwzfXH dZqHTV"
    href="/about_trusting_social/#about-trusting-social"
    level={2}
    onClick={[Function]}
    title="About Trusting Social"
  >
    <span
      className="sc-htpNat cIuRgo"
    >
      About Trusting Social
    </span>
  </a>
  <a
    className="sc-bwzfXH dZqHTV"
    href="/about_trusting_social/#about-the-credit-insight-service"
    level={2}
    onClick={[Function]}
    title="About the Credit Insight Service"
  >
    <span
      className="sc-htpNat cIuRgo"
    >
      About the Credit Insight Service
    </span>
  </a>
  <a
    className="sc-bwzfXH dZqHTV"
    href="/about_trusting_social/#encryption-policy"
    level={2}
    onClick={[Function]}
    title="Encryption Policy"
  >
    <span
      className="sc-htpNat cIuRgo"
    >
      Encryption Policy
    </span>
  </a>
  <a
    className="sc-bwzfXH igNvNI"
    href="/about_trusting_social/#encrypting-the-credit-insight"
    level={3}
    onClick={[Function]}
    title="Encrypting the Credit Insight"
  >
    <span
      className="sc-htpNat cIuRgo"
    >
      Encrypting the Credit Insight
    </span>
  </a>
  <a
    className="sc-bwzfXH eevfLr"
    href="/demo/"
    level={1}
    onClick={[Function]}
    title="Demo"
  >
    <span
      className="sc-htpNat eqFeY"
    >
      Demo
    </span>
    <svg
      className="sc-bdVaJa jjLnhl"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </a>
  <a
    className="sc-bwzfXH eevfLr"
    href="/customer_apis/"
    level={1}
    onClick={[Function]}
    title="Customer API"
  >
    <span
      className="sc-htpNat eqFeY"
    >
      Customer API
    </span>
    <svg
      className="sc-bdVaJa jjLnhl"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </a>
</div>
`;
